File: Bronze1-1/06-22-pm/unit06/max-work-week/week.cpp
/*
 * Max work week
 */
#include <bits/stdc++.h>
using namespace std;

int main() {
	int N; cin >> N;
	int M[N];
	for(int j=0; j<N; j++)
		cin >> M[j];
	
	int sum = 0;
	int maxM = 0;
	for(int x : M) {
		sum += x;
		maxM = max(maxM, x);
	}
	
	if(maxM*2<=sum)
		cout << sum << '\n';
	else
		cout << 2*(sum-maxM)+1 << '\n';
}

File: Bronze1-1/06-22-pm/week.java
/*
 * Max work week
 */
import java.util.*;
public class week {
	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		int N = in.nextInt();
		int[] M = new int[N];
		for(int j=0; j<N; j++)
			M[j] = in.nextInt();
		in.close();
		
		int sum = 0;
		int maxM = 0;
		for(int x : M) {
			sum += x;
			maxM = Math.max(maxM, x);
		}
		
		if(maxM*2<=sum)
			System.out.println(sum);
		else
			System.out.println(2*(sum-maxM)+1);
	}
}